src/core/layout/qgslayoutitemlabel.cpp

```diff
@@ -314,7 +328,25 @@ void QgsLayoutItemLabel::replaceDateText( QString &text ) const
 
 void QgsLayoutItemLabel::setFont( const QFont &f )
 {
-  mFont = f;
+  Q_NOWARN_DEPRECATED_PUSH
+  mFormat.setFont( f );
+  if ( f.pointSizeF() != -1 )
+    mFormat.setSize( f.pointSizeF() );
```

Comment:
@nyalldawson I have added this here and I was debating if I should add it in the QgsTextFormat instead. This is notably required for the label tests as the QFont has the size specified. And this doesn't break the current behavior.